Scripts/Generator/MapLoader.cs

```diff
@@ -0,0 +1,162 @@
+using Godot;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+public class MapLoader : Spatial
+{
+    [Export]
+    public int UnitSize
+    {
+        get
+        {
+            return unitSize;
+        }
+        set
+        {
+            unitSize = value;
+        }
+    }
+
+    [Export(PropertyHint.File)]
+    public String MapPath
+    {
+        get
+        {
+            return mapPath;
+        }
+        set
+        {
+            mapPath = value;
+        }
+    }
+
+    [Export]
+    public Material WallMaterial
+    {
+        get
+        {
+            return wallMaterial;
+        }
+        set
+        {
+            wallMaterial = value;
+        }
+    }
+
+    [Export]
+    public Material FloorMaterial
+    {
+        get
+        {
+            return floorMaterial;
+        }
+        set
+        {
+            floorMaterial = value;
+        }
+    }
+
+    private class Bounds
+    {
+        public int left, right, top, bottom;
+    }
+
+    private int unitSize;
```

Comment:
`unitSize` , `wallMaterial` and `floorMaterial` are not used outside of the getter and setter. I suggest creating a `RebuildMap()` which rebuilds the same map with the new properties.